### PR TITLE
Feature/upgrade dependencies

### PR DESCRIPTION
**Pull request by dl-ons**
**What**

The `https://github.com/ONSdigital/dp-rchttp`dependency is now deprecated. The `github.com/ONSdigital/dp-net/v2(http://github.com/ONSdigital/dp-net/v2` dependency appears to be available for `dp-authorisation` to use in the `v1` branch.

Tested this with a dependency tidy in https://github.com/ONSdigital/dp-authorisation/pull/75.

**How to review**

Check an application using `github.com/ONSdigital/dp-authorisation/auth` like https://github.com/ONSdigital/dp-search-api/pull/268 depends on `dp-net/v2` rather than `dp-rchttp` amongst other tests.

**Who can review**

Team Open Sauce.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/ONSdigital/dp-authorisation
 
-go 1.13
+go 1.21
```
Comment (lindenmckenzie) [inline go.mod:3]: believe we're on 1.22 :)

File: go.mod [hunk 2]
```diff
@@ -9,10 +9,79 @@ go 1.13
...
+	github.com/gorilla/mux v1.8.1
+	github.com/smartystreets/goconvey v1.8.1
+)
+
+require (
+	github.com/ONSdigital/dp-api-clients-go v1.43.0 // indirect
+	github.com/ONSdigital/dp-frontend-models v1.1.0 // indirect
+	github.com/ONSdigital/dp-healthcheck v1.6.1 // indirect
+	github.com/ONSdigital/dp-mocking v0.10.1 // indirect
+	github.com/ONSdigital/dp-net v1.5.0 // indirect
```
Comment (lindenmckenzie) [inline go.mod:23]: that's a pain, what's bringing that in?

File: go.mod [hunk 3]
```diff
@@ -9,10 +9,79 @@ go 1.13
...
+	github.com/ONSdigital/dp-net/v2 v2.11.2 // indirect
+	github.com/ONSdigital/go-ns v0.0.0-20191104121206-f144c4ec2e58 // indirect
```
Comment (lindenmckenzie) [inline go.mod:25]: that's a pain, what's bringing that in?

**Review by lindenmckenzie — COMMENTED** [review]
Minor comments.

File: go.mod (hunk 2 above)
Comment (dl-ons) [reply go.mod:23]: Commit `Remove github.com/ONSdigital/dp-rchttp` brings in:

- `github.com/ONSdigital/dp-frontend-models v1.1.0 // indirect`

Commit `Add github.com/ONSdigital/dp-api-clients-go/v2/headers package` brings in:

- `github.com/ONSdigital/dp-healthcheck v1.6.1 // indirect`
- `github.com/ONSdigital/dp-mocking v0.10.1 // indirect`
- `github.com/ONSdigital/dp-net v1.5.0 // indirect`

**Pull request closed without merging**